### PR TITLE
Introduce sid::unknown_kind

### PR DESCRIPTION
**Pull request by anstaf**

### CODE REVIEW

**Comment by anstaf** [comment]
launch jenkins

File: include/gridtools/sid/unknown_kind.hpp [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+namespace gridtools {
+    namespace sid {
+        // If for some SID `sid_get_strides_kind(SID const&)` returns `unknown_kind` `sid_get_strides_kind`,
```
Comment (havogt) [inline include/gridtools/sid/unknown_kind.hpp:14]:
```suggestion
        // If for some SID `sid_get_strides_kind(SID const&)` returns `unknown_kind`,
```

File: include/gridtools/sid/unknown_kind.hpp [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        // it has the special meaning: SID implementation does not guaranties that the strides are the same
```
Comment (havogt) [inline include/gridtools/sid/unknown_kind.hpp:15]:
```suggestion
        // it has the special meaning: SID implementation does not guarantee that the strides are the same
```

**Review by havogt — APPROVED** [review]
Looks good.

**Comment by anstaf** [comment]
launch jenkins

**Pull request merged**